### ISSUE TITLE
The training process  and results

### ISSUE THREAD

**Issue by zhou22222** [YongwonJo/SimRegMatch#1]
Hi, following the guide in paper , and i have runned the project successfully with the agedb dataset, but i found that my training process and result can not match the result in the paper. No matter how i change the hyperparameters, the MAE always sustained in 24.xxx and R^2 always sustained in -0.xxx. Is there any problem in the program? Actually, in the first time, loss is so big that the total loss is higher than 1000 and loss finally converges to 800+. 

![processing](https://github.com/user-attachments/assets/b5a448e1-87cb-4586-b515-a8a3e691e6b3)

**Comment by xueqings** [comment]
I try the method, some details trouble me.

In the code, the authod set percentile=0.95,
self.args.threshold = np.percentile(v_uncertainty.detach().cpu().numpy(), q=self.args.percentile)

The question is :
The meaning of the article is that the threshold is the KTH percentile of the variance, according to the code here, Percentile should be set to 95 rather than 0.95, if it is 0.95, then only the smallest of the first 0.95% can pass the variance screening, which is not very reasonable.

**Comment by zhou22222** [comment]
emmm，i think this is not the real problem of the model. Actually, the main problem, which causes that the model can not learn anything from the arthor's training strategy, is that the model uses the same image to compute for several times(arthor sets 5) without updating model params in one iteration. In my view, the result of computing several times wouldn't change(maybe some small changes because the random numpy), so the variance will be very very small following the results and the model can not select any pesudo-lables(they are always the 'same'). The arthor didn't give any pretrained model files.So, my friend, donot waste time on this one. maybe freematch, softmatch and so on are what you want. By the way, are you from china? Your name called xueqings, seems like a chinese name.



---- Replied Message ----
| From | ***@***.***> |
| Date | 08/15/2024 17:38 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [YongwonJo/SimRegMatch] The training process and results (Issue #1) |

I try the method, some details trouble me.

In the code, the authod set percentile=0.95,
self.args.threshold = np.percentile(v_uncertainty.detach().cpu().numpy(), q=self.args.percentile)

The question is :
The meaning of the article is that the threshold is the KTH percentile of the variance, according to the code here, Percentile should be set to 95 rather than 0.95, if it is 0.95, then only the smallest of the first 0.95% can pass the variance screening, which is not very reasonable.

**Comment by xueqings** [comment]
yeah,I am from China.Maybe you can finish training so as to see the results, I tried the imdb_wiki dataset, the final R2 is around 0.37, then I used my own code on the UTKFace dataset overlay this method, the results are below, you can refer to:

the loss:
![image](https://github.com/user-attachments/assets/803f8481-de0e-42a8-aedf-0b550b81e4f9)

the threshold:
![image](https://github.com/user-attachments/assets/f0419ccf-ab26-414b-af37-7a7c0477c9bd)

**Comment by zhou22222** [comment]
啊？我训完了，那个issue上我贴了一张训练图，收敛不了，loss什么的很高，请问你有修改代码吗？



---- Replied Message ----
| From | ***@***.***> |
| Date | 08/16/2024 09:53 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [YongwonJo/SimRegMatch] The training process and results (Issue #1) |

yeah,I am from China.Maybe you can finish training so as to see the results, I tried the imdb_wiki dataset, the final R2 is around 0.37, then I used my own code on the UTKFace dataset overlay this method, the results are below, you can refer to:

the loss:
image.png (view on web)

the threshold:
image.png (view on web)

**Comment by xueqings** [comment]
![image](https://github.com/user-attachments/assets/f2e7477e-acdc-4e12-8e93-5bdcc95e93ef)
这里的代码在更新模型参数，我在imdb_wiki数据集上跑了一下，没怎么修改代码的整体逻辑

**Comment by zhou22222** [comment]
我和你跑的不是一个数据集，我想起来了。这里确实是在更新，但是他在算那个方差的时候，用的是没有变化的模型反复算的一张图的结果



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年08月16日 09:59 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [YongwonJo/SimRegMatch] The training process and results (Issue #1) |

image.png (view on web)
这里的代码在更新模型参数，我在imdb_wiki数据集上跑了一下，没怎么修改代码的整体逻辑

**Comment by zhou22222** [comment]
我是在agedb上跑的



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年08月16日 09:59 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [YongwonJo/SimRegMatch] The training process and results (Issue #1) |

image.png (view on web)
这里的代码在更新模型参数，我在imdb_wiki数据集上跑了一下，没怎么修改代码的整体逻辑

**Comment by xueqings** [comment]
![image](https://github.com/user-attachments/assets/ff40a188-cc20-4abb-ab0b-bb303bc7112a)
这个是我跑的具体结果，这里我还设置错了把百分数的值设置为0.95，改成5或者95应该会好一点，方差不是通过多次得到的预测值在变化嘛，模型在更新，方差也在更新啊

**Comment by zhou22222** [comment]
参数不变下，模型预测同一个数据应该没啥变化。具体在他train（）函数下，有一个循环，那个循环就是反复算同一个数据的结果，算完这个才去算的方差。



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年08月16日 10:08 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [YongwonJo/SimRegMatch] The training process and results (Issue #1) |

image.png (view on web)
这个是我跑的具体结果，这里我还设置错了把百分数的值设置为0.95，改成5或者95应该会好一点，方差不是通过多次得到的预测值在变化嘛，模型在更新，方差也在更新啊

**Comment by xueqings** [comment]
模型加了dropout,预测结果怎么会一样

**Comment by zhou22222** [comment]
确实加了，我设置的0.1，在我debugging时候，我发现我的变化只在前几轮中，后边没什么太大改变



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年08月16日 10:14 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [YongwonJo/SimRegMatch] The training process and results (Issue #1) |

模型加了dropout,预测结果怎么会一样

**Comment by zhou22222** [comment]
待会给你看下我的训练结果，agedb上300epoch，最后r2是超级小的负数emmm



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年08月16日 10:14 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [YongwonJo/SimRegMatch] The training process and results (Issue #1) |

模型加了dropout,预测结果怎么会一样
